client/my-sites/plugins/with-plugin-redirect.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import isSiteAutomatedTransfer from 'calypso/state/selectors/is-site-automated-transfer';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import { isJetpackSite, getSiteWoocommerceWizardUrl } from 'calypso/state/sites/selectors';
+import { isPluginActionCompleted } from 'calypso/state/plugins/installed/selectors';
+import { INSTALL_PLUGIN } from 'calypso/lib/plugins/constants';
+import getSiteConnectionStatus from 'calypso/state/selectors/get-site-connection-status';
+import { getAutomatedTransferStatus } from 'calypso/state/automated-transfer/selectors';
+import { transferStates } from 'calypso/state/automated-transfer/constants';
+import { successNotice, removeNotice } from 'calypso/state/notices/actions';
+
+// Plugins list that should perfomr a redirect
+// once it's installed.
+const redirectingPluginsList = {
+	woocommerce: getSiteWoocommerceWizardUrl,
+};
+
+/**
+ * Return the handler function to get the plugin redirect URL.
+ *
+ * @param {string} slug - Plugin slug.
+ * @returns {Function|boolean} Selector handler if the plugin should redirect. Otherwise, False.
+ */
+function getPluginRedirectUrl( slug ) {
+	if ( Object.keys( redirectingPluginsList ).indexOf( slug ) < 0 ) {
+		return false;
+	}
+
+	return redirectingPluginsList[ slug ];
+}
+
+const withPluginRedirect = ( Component ) => ( props ) => {
```

Comment:
As this is using hooks internally, should we change it to a custom hook? `usePluginRedirect` perhaps?

Reply:
yes, we can. Let me take a look.

Reply:
Let's tackle it in a follow-up PR. The main reason why we use a HOC is that the `<PluginMeta />` is not a functional component